{{cookiecutter.project_slug}}/server/{{cookiecutter.project_slug}}/core/serializers.py

```diff
@@ -68,3 +74,21 @@ def validate(self, data):
 
     def create(self, validated_data):
         return User.objects.create_user(**validated_data)
+
+{%- if cookiecutter.include_mobile == 'y' %}
+class CodeResetPasswordSerializer(serializers.Serializer):
+    code = serializers.CharField(allow_blank=False, required=True)
+    password = serializers.CharField(allow_blank=False, required=True)
+
+    def validate_code(self, value):
+        code_from_db = self.context.get("user").reset_password_codes.filter(is_deactivated=False).first()
+        if not code_from_db or not code_from_db.is_valid or not check_password(str(value), code_from_db.code):
```

Comment:
Here we are first filtering for `is_deactivated=False` and then we check the `is_valid` which also checks that.

Maybe we could add a method in `UserResetPasswordCodeManager` to fetch only valid codes:

```py
class UserResetPasswordCodeManager(models.Manager):
    ...
    def filter_valid():
        return self.filter(is_deactivated=False, is_used=False)
    ...
```

Or maybe the `check_password` should also be in the `is_valid` method?

Reply:
actually the is valid is supposed to also check that it hasn't expired, I thought I had that in there but it appears to be missing :( either way you are right we can condense this into a single filter 

```
def filter_valid():
           return self.filter(is_deactivated=False, is_used=False, datetime_created__gte=timezone.now()-timedelta(minutes=x))
```